Src/DependencyCollector/Shared/Implementation/DependencyDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,308 @@
+                        }
+                    case "operation.data": // not defined by OpenTracing
+                        {
+                            telemetry.Data = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "db.statement":
+                        {
+                            dbStatement = tag.Value;
+                            break;
+                        }
+                    case "error":
+                        {
+                            bool failed;
+                            if (bool.TryParse(tag.Value, out failed))
+                            {
+                                telemetry.Success = !failed;
+                                continue; // skip Properties
+                            }
+                            break;
+                        }
+                    case "http.status_code":
+                        {
+                            telemetry.ResultCode = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "http.method":
+                        {
+                            httpMethodWithSpace = tag.Value + " ";
+                            break;
+                        }
+                    case "http.url":
+                        {
+                            httpUrl = tag.Value;
+                            // if no valid URI specified in the payload use this one
+                            if (requestUri == null && !string.IsNullOrEmpty(tag.Value))
+                            {
+                                if (Uri.TryCreate(tag.Value, UriKind.RelativeOrAbsolute, out requestUri))
+                                {
+                                    continue; // skip Properties
+                                }
+                            }
+                            break;
+                        }
+                    case "peer.address":
+                        {
+                            peerAddress = tag.Value;
+                            break;
+                        }
+                    case "peer.hostname":
+                        {
+                            telemetry.Target = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "peer.service":
+                        {
+                            telemetry.Type = tag.Value;
+                            continue; // skip Properties
+                        }
+                }
+
+                // if more than one tag with the same name is specified, the first one wins
+                if (!telemetry.Context.Properties.ContainsKey(tag.Key))
+                {
+                    telemetry.Context.Properties[tag.Key] = tag.Value;
+                }
+            }
+
+            telemetry.Context.Properties["activity"] = currentActivity.OperationName;
```

Comment:
in this implementation "type" comes from "peer.service" tag and in some cases it will be a display string (with whitespace). On the other hand we should have activity (and DS) name available so that customers can identify noisy data sources